client/my-sites/people/contractor-select/index.tsx

```diff
@@ -0,0 +1,50 @@
+/**
+ * External dependencies
+ */
+import React, { FunctionComponent } from 'react';
+import { useTranslate } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import FormCheckbox from 'components/forms/form-checkbox';
+import FormFieldset from 'components/forms/form-fieldset';
+import FormLabel from 'components/forms/form-label';
+import SupportInfo from 'components/support-info';
+
+interface Props {
+	checked: boolean;
+	onChange: ( event ) => void;
```

Comment:
we don't _need_ to lock this down so much; if we want to be permissive but safe then we can constrain the arguments and leave the return type open with a type parameter

```ts
onChange: <T>( event ) => T;
```

or get even more specific, if it's warranted…
```ts
onChange: <T>( event: ChangeEvent ) => T;
```

I tried getting the type of the inner `onChange` from `FormCheckbox` but it's a little gray here what to do. That type is a little elusive to me on how to extract it and also at some point we're wanting to enforce this upwards on the callers.

The risk in typing `ChangeEvent` is that `FormCheckbox` could change - I don't know if that interface is intentionally set with `ChangeEvent` or not. The risk in setting `any` obviously is the lack of safety.

Once `FormCheckbox` is typed this will be better. A third option is that for now we could also assign it as `Function`.

Take your pick, or leave it as is - I just wanted to chime in with some extra context here in case you hadn't already considered it.

Reply:
I was curious and looked around.

TypeScript definitions don't seem to have an `ChangeEvent`, `onchange` just uses `Event` and they're defined as `onchange: ((this: [THE_THIS_TYPE], ev: Event) => any) | null`.

@dmsnell were you suggesting omitting the type here for now? That seems appealing. `FormCheckbox` is closer to a leaf that should be simple to type. At that point types here can likely be inferred or extracted from `FormCheckbox`.

Reply:
`@types/react` defines the change event in [@types/react/index.d.ts](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/7ff648e68bbb1d4fdedcc8efcfe9e0c8ff4d0d45/types/react/index.d.ts#L1943) as `onChange?: ChangeEventHandler<T>;` and it's the `<input>` element which is typed with a `<T>` - that's the part I had trouble getting through.

To type it _through the system_ we need `form-checkbox` to be parameterized and get that into `contractor-select/index.tsx` which probably should be trivial and I'm likely missing something obvious.

> were you suggesting omitting the type here for now?

either omitting it altogether or parameterizing it so we don't require a void function return. no harm done if the function returns something and that's ignored